lib/Client.js

```diff
@@ -339,17 +339,27 @@ class Client extends EventEmitter {
     * Ban a user from a guild
     * @arg {String} guildID The ID of the guild
     * @arg {String} userID The ID of the user
-    * @arg {Number} [deleteMessageDays=0] Number of days to delete messages for, between 0-7 inclusive
-    * @arg {String} [reason] The reason to be displayed in audit logs
+    * @arg {Number} [options.deleteMessageDays=0] [DEPRECATED] Number of days to delete messages for, between 0-7 inclusive
+    * @arg {Number} [options.deleteMessageSeconds=0] Number of seconds to delete messages for, between 0 and 604800 inclusive
+    * @arg {String} [options.reason] The reason to be displayed in audit logs
+    * @arg {String} [reason] [DEPRECATED] The reason to be displayed in audit logs
     * @returns {Promise}
     */
-    banGuildMember(guildID, userID, deleteMessageDays, reason) {
-        if(!isNaN(deleteMessageDays) && (deleteMessageDays < 0 || deleteMessageDays > 7)) {
-            return Promise.reject(new Error(`Invalid deleteMessageDays value (${deleteMessageDays}), should be a number between 0-7 inclusive`));
+    banGuildMember(guildID, userID, options, reason) {
+        if(!options || typeof options !== "object") {
+            options = {
+                deleteMessageDays: options
+            };
+        }
+        if(reason !== undefined) {
+            options.reason = reason;
+        }
+        if(options.deleteMessageDays && !isNaN(options.deleteMessageDays) && !Object.hasOwn(options, "deleteMessageSeconds")) {
```

Comment:
```suggestion
        if(options.deleteMessageDays && !isNaN(options.deleteMessageDays) && !options.hasOwnProperty("deleteMessageSeconds")) {
```

`Object.hasOwn()` was added in 16.9.0. 100% in favour of switching to 16.9.0, just wanted to point this out as we're still on 10.40.